skorch/callbacks/scoring.py

```diff
@@ -46,11 +46,16 @@ def convert_sklearn_metric_function(scoring):
     sklearn scorer and return it. Otherwise, return ``scoring`` unchanged."""
     if callable(scoring):
         module = getattr(scoring, '__module__', None)
+
+        # those are scoring objects returned by make_scorer starting
+        # from sklearn 0.22
+        scorer_names = ('_PredictScorer', '_ProbaScorer', '_ThresholdScorer')
         if (
                 hasattr(module, 'startswith') and
                 module.startswith('sklearn.metrics.') and
                 not module.startswith('sklearn.metrics.scorer') and
-                not module.startswith('sklearn.metrics.tests.')
+                not module.startswith('sklearn.metrics.tests.') and
+                not scoring.__class__.__name__ in scorer_names
```

Comment:
Is the `scoring.__class__.__name__` check needed? `make_scorer` returned these objects in 0.21 as well.

Reply:
I don't quite understand your suggestion. We need this new line to to check if we get any of these objects. I tried to remove the new line and get failing tests with sklearn 0.22 but not 0.21.